staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -185,6 +191,18 @@ func (t *tracker) List(gvr schema.GroupVersionResource, gvk schema.GroupVersionK
 	return list.DeepCopyObject(), nil
 }
 
+func (t *tracker) Watch(gvr schema.GroupVersionResource, ns string) (watch.Interface, error) {
```

Comment:
This can end up called in parallel from multiple threads, right?  Please lock it.  Using the big lock will do.

Reply:
I just realise that current implementation of `Watch` call doesn't support multi-thread invocation on the same namespace of the same resource. Because every namespace of every resource has only one `watcher` which has one watch event channel notifying the changes. In the fact, complete multi-threading support can be sort of complicated for a fake client, because it's hard to let `Watch` know if itself is called from a different go routine or a different context. btw the fakeclient lacks the implementation of `Limit/Continue` support of `List` call as well, I will open another issue to discuss about it.

Reply:
A simple watch "broadcasting" may help, I'm not sure if it is necessary. WDYT?

Reply:
> I just realise that current implementation of Watch call doesn't support multiple invocation on the same namespace of the same resource.

Good catch.  I think that you'll need to have a `map[GroupVersionResource]map[namespace][]watch.Interface`.